File: lib/arc4random.c
#include <stdint.h>
#include <time.h>


static uint32_t simple_rand(void) {
    static uint32_t seed = 0;
    seed = (seed * 1103515245 + 12345) & 0x7FFFFFFF;
    return seed;
}

void arc4random_buf(void *buf, size_t nbytes) {
    uint8_t *ptr = buf;
    size_t i;

    for (i = 0; i < nbytes; i++) {
        ptr[i] = simple_rand() & 0xFF;
    }
}

File: Sources/lib/arc4random.swift

func simpleRand() -> UInt32 {
    var seed: UInt32 = 0
    seed = (seed &* 1103515245 &+ 12345) & 0x7FFFFFFF
    return seed
}

@_cdecl("arc4random_buf")
public func arc4random_buf(_ buf: UnsafeMutableRawPointer!, _ nbytes: Int) {
    let ptr = buf.bindMemory(to: UInt8.self, capacity: nbytes)
    for i in 0..<nbytes {
        ptr[i] = UInt8(simpleRand() & 0xFF)
    }
}